ui/src/BaseEditor.ts

```diff
@@ -0,0 +1,166 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { UserCancelledError } from './errors';
+import { localize } from "./localize";
+import { TemporaryFile } from './utils/TemporaryFile';
+
+// tslint:disable-next-line:typedef
+const dialogResponses = {
+    OK: localize('OK', "OK"),
+    DontShowAgain: localize('dontShow', "Don't Show Again")
+};
+
+// tslint:disable-next-line:no-unsafe-any
+export abstract class BaseEditor<ContextT> implements vscode.Disposable {
+    private fileMap: { [key: string]: [vscode.TextDocument, ContextT] } = {};
+    private ignoreSave: boolean = false;
+
+    constructor(readonly showSavePromptKey: string, readonly outputChanel?: vscode.OutputChannel) {

```

Comment:
Just my two cents, but I don't like parameter properties because it's easier for people to use incorrect signatures. For example, both of these should be private, correct?

I don't think people would've made that mistake if they were forced to write properties separately.

Reply:
This is regarding the tslint rules, right?  

I remember this being discussed with the App Services Extension.  I thought we decided to disable this rule because people didn't like being forced to define the property and then assigning it in the constructor.